sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1304,6 +1304,15 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       assert(rlike3.count() == 0)
     }
   }
+
+  test("SPARK-21538: Attribute resolution inconsistency in Dataset API") {
+    val df = spark.range(1).withColumnRenamed("id", "x")
+    checkAnswer(df.sort(col("id")), df.sort("id"))
+    checkAnswer(df.sort($"id"), df.sort("id"))
+    checkAnswer(df.sort('id), df.sort("id"))
+    checkAnswer(df.orderBy('id), df.sort("id"))
+    checkAnswer(df.orderBy("id"), df.sort("id"))
```

Comment:
Since it sounds like you are very interested in Spark open source contribution, let me explain how we write the test cases. Normally, we do not want to duplicate the codes and also want to verify the results. Thus, maybe you can change the code like

```Scala
    val df = spark.range(3).withColumnRenamed("id", "x")
    val expected = Row(0) :: Row(1) :: Row (2) :: Nil
    checkAnswer(df.sort("id"), expected)
    checkAnswer(df.sort(col("id")), expected)
    checkAnswer(df.sort($"id"), expected)
    checkAnswer(df.sort('id), expected)
    checkAnswer(df.orderBy('id), expected)
    checkAnswer(df.orderBy("id"), expected)
```